### PR TITLE
Bugfix: Fix issue handling two dots preceding line

### PR DESCRIPTION
**Pull request by timbess**
When there is a space or tab at the beginning of a line, yenc handles that by prepending two dots to the beginning of the line. We need to handle that case in `decode_buffer`.

### CODE REVIEW

File: src/decode.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ use std::path::PathBuf;
 
 use errors::DecodeError;
 use crc32;
-use constants::{CR, DEFAULT_LINE_SIZE, ESCAPE, LF, NUL, SPACE};
+use constants::{CR, DEFAULT_LINE_SIZE, ESCAPE, LF, NUL, SPACE, DOT};
```
Comment (aswaving) [inline src/decode.rs]: Please use rustfmt, or directly put DOT after DEFAULT_LINE_SIZE.

File: src/decode.rs [hunk 2]
```diff
@@ -126,17 +126,29 @@ where
...
-    while let Some(byte) = iter.next() {
+    let mut iter = input.iter().enumerate();
+    while let Some((col, byte)) = iter.next() {
         let mut byte = *byte;
         match byte {
             NUL | CR | LF => {
                 // for now, just continue
                 continue;
             }
+            DOT => {
```
Comment (aswaving) [inline src/decode.rs:137]: I'm missing code coverage for this new code. Please add a unit test for this.

**Review by aswaving — CHANGES_REQUESTED** [review]

**Comment by timbess** [comment]
Ok cool, I ran rustfmt on the file and added a test case.

File: src/decode.rs [hunk 3]
```diff
@@ -466,4 +465,11 @@ mod tests {
         );
     }
 
+    #[test]
```
Comment (aswaving) [inline src/decode.rs:468]: Great, thanks for this test case. However, and sorry for being picky, decoding &[b'.'] and &[b'.', x] (where x is some byte other than b'.') are not covered yet. Could you provide a test case?

File: src/decode.rs [hunk 4]
```diff
@@ -466,4 +465,11 @@ mod tests {
...
+    fn decode_valid_prepended_dots() {
+        assert_eq!(
+            &vec![b'.' - 0x2A],
+            &decode_buffer(&b".."[..]).unwrap()
```
Comment (aswaving) [inline src/decode.rs]: Are you sure you need to copy the data ([..])? I think you can leave it out,`decode_buffer(b"..")` seems sufficient.
The alternative would be `decode_buffer(&[b'.', b'.'])`.

Comment (timbess) [reply src/decode.rs]: Oh I don't think it copies, that just explicitly creates a slice. But I removed the [..] and it worked! I could swear it created a compile error for comparing an array with a slice before. ¯\_(ツ)_/¯

File: src/decode.rs (hunk 3 above)
Comment (timbess) [reply src/decode.rs:468]: yeah that's fine

File: src/decode.rs (hunk 4 above)
Comment (aswaving) [reply src/decode.rs]: You're right. This doesn't copy, I mixed up Python's [:] with Rust's [..] :-(

**Review by aswaving — APPROVED** [review]

**Pull request merged**